fastplotlib/layouts.py

```diff
@@ -102,18 +110,32 @@ def __init__(
             camera = cameras[i, j]
             controller = self._controllers[i, j]
 
-            self.subplots[i, j] = Subplot(
+            if self.names is not None:
+                name = self.names[i, j]
+            else:
+                name = None
+
+            self._subplots[i, j] = Subplot(
                 position=position,
                 parent_dims=(nrows, ncols),
                 camera=camera,
                 controller=controller,
                 canvas=canvas,
-                renderer=renderer
+                renderer=renderer,
+                name=name
             )
 
         self._animate_funcs: List[callable] = list()
         self._current_iter = None
 
+    def __getitem__(self, index: Union[Tuple[int, int], str]):
+        if type(index) == str:
+            for subplot in self._subplots.ravel():
+                if subplot.name == index:
+                    return subplot
```

Comment:
After this return can you raise IndexError for "no Subplot with given name" or something